### PR TITLE
Combine simple string filter and string filter

### LINKED ISSUE #1932: [Program Merge] Consolidate string filter
Carry over from this PR: https://github.com/openmsupply/open-msupply/pull/1882

https://github.com/openmsupply/open-msupply/pull/1882#discussion_r1236318649

Comments: 

Consolidate StringFilter, SimpleStringFilter and EqualFilter<String> into one.

There is an argument that we may not want to expose some expensive filters for fields where it doesn't make sense, like UUID field. I think in this case it's less logic to just have one type of filter, and reduces overhead of determining which fields should use which filter (you can argue if too many filters are exposed for a field, then we are passing on overhead to the consumer, i think that's better then having three different types, that seem to be slowly converging anyways)

`Update`: Maybe consolidate SimpleStringFilter and StringFilter, and keep EqualFilter<String> for things like ids

### PR DESCRIPTION
**Pull request by andreievg**
Fixes #1932

# 👩🏻‍💻 What does this PR do? 

Use one common string filter type, most changes are find and replace apart from (see self review comment in code changes for those):

server/repository/src/diesel_macros.rs
server/graphql/core/src/generic_filters.rs
server/repository/src/db_diesel/filter_sort_pagination.rs
server/service/src/programs/patient/search.rs

# 🧪 How has/should this change been tested? 

Not entirely sure which areas might be affected, I did unit test, seem to pass. Also strict typing should pretty much that these changes are safe. 

Perhaps it's worth checking that case insensitive filter still work (searching patient by first/last name is the same as it was)

## 📃 Documentation

### CODE REVIEW

File: server/repository/src/diesel_macros.rs
Comment (andreievg) [inline server/repository/src/diesel_macros.rs:1]: Previously in this file there was apply_simple_string_filter_method, apply_simple_string_filter and simple_string_or_filter. They were basically replaces with apply_string_filter_method, apply_string_filter and apply_string_or_filter

File: server/graphql/core/src/generic_filters.rs [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
+use repository::{DateFilter, DatetimeFilter, EqualFilter, StringFilter};
 
 // simple string filter
 #[derive(InputObject, Clone)]
-pub struct SimpleStringFilterInput {
+pub struct StringFilterInput {
     /// Search term must be an exact match (case sensitive)
     equal_to: Option<String>,
-    /// Search term must be an exact match, but case insensitive
-    insensitive_equal_to: Option<String>,
```
Comment (andreievg) [inline server/graphql/core/src/generic_filters.rs:11]: I couldn't find any uses for this in front end. @clemens-msupply could it be use in reports ?

File: server/repository/src/db_diesel/filter_sort_pagination.rs [hunk 2]
```diff
@@ -1,40 +1,7 @@
 use chrono::{NaiveDate, NaiveDateTime};
 use util::inline_init;
 
-#[derive(Clone, PartialEq, Debug)]
```
Comment (andreievg) [inline server/repository/src/db_diesel/filter_sort_pagination.rs:4]: Replaced by StringFilter

File: server/service/src/programs/patient/search.rs [hunk 3]
```diff
@@ -37,16 +37,16 @@ pub fn patient_search(
     } = input;
 
     if let Some(code) = code {
-        filter = filter.code(SimpleStringFilter::equal_to(&code));
```
Comment (andreievg) [inline server/service/src/programs/patient/search.rs:40]: @clemens-msupply, previously insensitive_equal_to would be the same as like but sanitised (with '%' removed), was there a danger that user would be searching with '%' ? Security risk or too many matches ?

File: server/graphql/core/src/generic_filters.rs (hunk 1 above)
Comment (clemens-msupply) [reply server/graphql/core/src/generic_filters.rs:11]: Just checked, there is no use of it. Please feel free to remove.

**Review by mark-prins — APPROVED** [review]
👍 I've had a browse through and the changes look ok - have tested all of the filters that I can find on the site and they're behaving as they used to

**Comment by andreievg** [comment]
thank you for review !

**Pull request merged**